File: frontend/src/services/EmployeeService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/employees";

//View Employee Details
export const getEmployee = () => axios.get(API_URL);

//Add Employee Deyails
export const addEmployee = (emp) => {return axios.post(API_URL, emp);};

//Update Employee Details
export const updateEmployee = (id, emp)=>{
  return axios.put(`${API_URL}/${id}`, emp);
};

//Delete Employee Details
export const deleteEmployee = (id) =>axios.delete(`${API_URL}/${id}`);

//Get Employee By Id
export const getEmployeeById = (id) => {return axios.get(`${API_URL}/${id}`);};

//Search Employee By Name
export const searchEmployeeByName = (name) => {
  return axios.get(`${API_URL}/search?name=${name}`);
};

File: backend/src/main/java/com/example/employee/service/EmployeeService.java
package com.example.employee.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.employee.entity.Employee;
import com.example.employee.repository.EmployeeRepository;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;
    //Save Employee Details
    public Employee saveEmployee(Employee emp) {
        return employeeRepository.save(emp);
    }

    //View Employee List
    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }
    
    
// // Get By Id
    public Employee getEmployeeById(Long id) {
        return employeeRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Employee Not Found"));
    }
    
    
 // Update Logic
    public Employee updateEmployee(Long id, Employee emp) {

        Employee existing = employeeRepository.findById(id).orElseThrow();

        existing.setName(emp.getName());
        existing.setEmail(emp.getEmail());
        existing.setDepartment(emp.getDepartment());
        existing.setSalary(emp.getSalary());

        return employeeRepository.save(existing); // UPDATE
    }



 // Delete
    public void deleteEmployee(Long id) {
        Employee emp = employeeRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Employee Not Found"));

        employeeRepository.delete(emp);
    }
    
    //Search Employee by name
    public List<Employee> searchEmployeeByName(String name) {
        return employeeRepository.findByNameContainingIgnoreCase(name);
    }

}
